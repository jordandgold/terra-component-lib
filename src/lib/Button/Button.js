import React from "react";

import "./Button.scss";

const Button = props => {
  const { onClick, className, text } = props;

  return (
    <div>
      <button
        onClick={e => onClick(e)}
        className={`ter-button ter-button--${className}`}
      >
        {text}
      </button>
    </div>
  );
};

export default Button;
