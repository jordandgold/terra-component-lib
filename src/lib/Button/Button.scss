.ter-button {
  box-sizing: border-box;
  font-size: 16px;
  padding: 8px 16px;
  background-color: #363545;
  color: #f3f3f7;
}
