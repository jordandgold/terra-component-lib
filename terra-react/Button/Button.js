import React from "react";
import "./Button.scss";

var Button = function Button(props) {
  var _onClick = props.onClick,
      className = props.className,
      text = props.text,
      name = props.name;
  return React.createElement(React.Fragment, null, React.createElement("button", {
    onClick: function onClick(e) {
      return _onClick(e);
    },
    className: "ter-button ".concat(className),
    name: name
  }, text));
};

export default Button;