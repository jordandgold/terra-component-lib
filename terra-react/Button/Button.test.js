import React from "react";
import { shallow } from "enzyme";
import Button from "./Button";
describe("Button", function () {
  var wrapper;
  var mockOnClick;
  var mockEventObject;
  beforeEach(function () {
    mockOnClick = jest.fn();
    wrapper = shallow(React.createElement(Button, {
      name: "test name",
      className: "test class",
      onClick: mockOnClick,
      text: "test text"
    }));
  });
  it("should match the snapshot", function () {
    expect(wrapper).toMatchSnapshot();
  });
  it("should call onClick method when clicked", function () {
    mockEventObject = {
      target: {
        innerHTML: "hello",
        name: "test name"
      }
    };
    wrapper.find(".ter-button").simulate("click", mockEventObject);
    expect(mockOnClick).toHaveBeenCalledWith(mockEventObject);
  });
});