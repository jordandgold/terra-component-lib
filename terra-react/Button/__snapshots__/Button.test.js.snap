// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match the snapshot 1`] = `
<Fragment>
  <button
    className="ter-button test class"
    name="test name"
    onClick={[Function]}
  >
    test text
  </button>
</Fragment>
`;
